packages/cactus-plugin-ledger-connector-cdl-socketio/README.md

```diff
@@ -0,0 +1,81 @@
+# `@hyperledger/cactus-plugin-ledger-connector-cdl-socketio`
+
+This plugin provides `Cacti` a way to interact with Fujitsu CDL networks. Using this we can perform:
+
+- `sendSyncRequest`: Send sync-typed requests to the API.
+- `sendAsyncRequest`: Send async-typed requests to the API.
+
+## Getting started
+
+### Required software components
+
+- OS: Linux (recommended Ubuntu20.04,18.04 or CentOS7)
+- Docker (recommend: v17.06.2-ce or greater)
+- node.js v12 (recommend: v12.20.2 or greater)
+
+## Boot methods
+
+### Common setup
+
+1. Always run configure command first, from the project root directory:
+
+   ```bash
+   pushd ../..
+   npm run configure
+   popd
+   ```
+
+1. Copy default configuration
+
+- **Remember to replace default CA and to adjust the `default.yaml` configuration on production deployments!**
+  ```bash
+  mkdir -p /etc/cactus/connector-cdl-socketio
+  rm -r /etc/cactus/connector-cdl-socketio/*
+  cp -rf ./sample-config/* /etc/cactus/connector-cdl-socketio/
+  ```
+
+#### Configuring CDL API Gateway Access
+
+- Set the base URL of GW service in `cdlApiGateway.url`. Do not include `api/v1`, just the base URL. (example: `"http://localhost:3000"`).
+- If the service certificate is signed with a known CA (node uses Mozilla DB), then you can skip the next steps.
+- If the service is signed with unknown CA, you can specify the gateway certificate to trust manually:
+  - Set `cdlApiGateway.caPath` to path of API Gateway certificate (in PEM format). (example: `"/etc/cactus/connector-cdl-socketio/CA/cdl-api-gateway-ca.pem"`)
+  - (optional) If server name in cert doesn't match the one in `cdlApiGateway.url`, you can overwrite it in `cdlApiGateway.serverName`
+- (not recommended - only for development): To ignore certificate rejection (e.g. use self-signed certificate) set `cdlApiGateway.skipCertCheck` to `true`.
+
+### Docker
+
+- Docker build process will use artifacts from the latest build. Make sure `./dist` contains the version you want to dockerize.
+
+```
+# Build
+docker build . -t cactus-plugin-ledger-connector-cdl-socketio
+
+# Run
+docker run -v/etc/cactus/:/etc/cactus -p 5061:5061 cactus-plugin-ledger-connector-cdl-socketio
+```
+
+### Manual
+
+```
+npm run start
+```
+
+## Configuration
+
+- Validator can be configured in `/etc/cactus/connector-cdl-socketio/default.yaml` (see [sample-config](./sample-config/default.yaml) for details).
+
+## Manual Tests
+
+- `cdl-connector-manual.test` contains a Jest test script that will check every implemented operation on running CDL instance.
```

Comment:
@outSH I tried running the test with `$ yarn jest packages/cactus-plugin-ledger-connector-cdl-socketio/src/test/typescript/integration/cdl-connector-manual.test.ts` but it fails because you marked the test case to be ignored in the jest config. Is there any other way I can run the test?

Reply:
@petermetz Please read https://github.com/hyperledger/cacti/issues/2455. In short - you need access to real CDL environment, hence the "manual" in the test name. I've updated the readme description of it for more clarity.

Reply:
@outSH Understood, thanks for clarifying. I understood 'manual' in a way that as long as I'm willing to perform the needed manual labor (running some commands, installing dependencies, etc.) the test can run like any other test case in the project (for free).

I can accept this as-is, but I do want to ask: Do you have at least tentative future plans to make it so that the CDL has some freeware edition that can be used to run these tests without paying?